airbyte-integrations/connectors/destination-meilisearch/destination_meilisearch/writer.py

```diff
@@ -0,0 +1,47 @@
+#
+# Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+#
+
+import time
+from collections.abc import Mapping
+from uuid import uuid4
+
+from meilisearch import Client
+
+
+class MeiliWriter:
+    write_buffer = []
+    flush_interval = 50000
+
+    def __init__(self, client: Client, steam_name: str, primary_key: str):
+        self.client = client
+        self.steam_name = steam_name
+        self.primary_key = primary_key
+
+    def queue_write_operation(self, data: Mapping):
+        random_key = str(uuid4())
+        self.write_buffer.append({**data, self.primary_key: random_key})
+        if len(self.write_buffer) == self.flush_interval:
+            self.flush()
+
+    def flush(self):
+        if len(self.write_buffer) == 0:
+            return
+
+        response = self.client.index(self.steam_name).add_documents(self.write_buffer)
+        self.wait_for_job(response.task_uid)
+        self.write_buffer.clear()
+
+    def wait_for_job(self, task_uid: str):
+        timer = 1
+        while True:
+            time.sleep(timer)
+            if timer < 60:
+                timer = timer * 2
+            try:
+                task = self.client.get_task(task_uid)
+                status = task["status"]
+                if status == "succeeded" or status == "failed":
+                    break
+            except:
+                timer = 60
```

Comment:
```suggestion
    def wait_for_job(self, task_uid: str):
        timer = 1
        while timeout < timeout_limit:
            time.sleep(timer)
            timer = min(timer * 2, 60)
            task = self.client.get_task(task_uid)
            status = task["status"]
            if status == MeiliSearchTaskStatus.SUCCEDED or status == MeiliSearchTaskStatus.FAILED:
                break
```
Please simplify the function, if you're going to do a try/except you must raise the error you're dealing with.
Maybe take a look in the backoff library to apply the timer.
I think if the task == 'failed' you can log the reason why it failed.
Would be nice to have a timeout_limit to not have an infinity loop here.
Add unit tests to all functions for the writer.

Reply:
@marcosmarxm meilisearch sdk has a method to wait for the async operation to finished. Now is much simpler.
I added the unit tests and logger massage as well. 

The [destination docs](https://docs.airbyte.com/connector-development/tutorials/building-a-python-destination) refers the KvDB destination connector but it lacks unit tests and logger message on flush method.